vignettes/intro_opencage.Rmd.src

```diff
@@ -0,0 +1,190 @@
+
+By default, OpenCage will store your queries in its server logs and will cache the forward geocoding requests on their side.
+They do this in order to speed up response times and to be able to debug errors and improve their service. 
+Logs are automatically deleted after six months according to OpenCage's [page on data protection and GDPR](https://opencagedata.com/gdpr).
+
+If you have concerns about privacy and want OpenCage to have no record of your query, i.e. the placename or latitude/longitude coordinates you want to geocode, you can set a `no_record` parameter to `TRUE` and the query contents are no longer logged nor cached. 
+OpenCage still records that you made a request, but not the specific query you made. 
+
+`oc_config(no_record = TRUE)` sets an `oc_no_record` option for the active R session, so it will be used for all subsequent OpenCage queries. 
+You can set the `oc_no_record` option persistently across sessions in your `.Rprofile`.
+
+For more information on OpenCage's policies on privacy and data protection see the Legal section in [their FAQs](https://opencagedata.com/faq#legal), their [GDPR page](https://opencagedata.com/gdpr), and, for the `no_record` parameter specifically, see the relevant [blog post](https://blog.opencagedata.com/post/145602604628/more-privacy-with-norecord-parameter).
+
+Please note that the {opencage} package always caches the data it receives from the OpenCage API, but only for as long as your R session is alive (see below).
+
+### Altogether now
+
+In sum, if you want to set your API key with {keyring}, set the rate limit to 10 (only do this if you have a paid account, please!), and do not want OpenCage to have records of your queries, you would configure {opencage} for the active session like this:
+
+```{r oc_config, eval=FALSE}
+library("opencage")
+oc_config(
+  key = keyring::key_get("opencage"),
+  rate_sec = 10,
+  no_record = TRUE
+)
+```
+
+## Forward geocoding
+
+Now you can start to geocode. Forward geocoding is from placename to latitude and longitude tuple(s).
+
+```{r forward}
+oc_forward_df(placename = "Sarzeau")
+```
+
+All geocoding functions are vectorised, i.e. you can geocode multiple placenames with one function call, even though the requests are still sent to and results received from the API one-by-one.
+
+```{r vectorised}
+opera <- c("Palacio de Bellas Artes", "Scala", "Sydney Opera House")
+oc_forward_df(placename = opera)
+```
+
+By default, `oc_forward_df()` only returns three results columns: `oc_lat` (for latitude), `oc_lon` (for longitude), and  `oc_formatted` (the formatted address). 
+As you can see, the results columns are all prefixed with `oc_`. 
+If you specify `oc_forward_df(output = all)`, you will receive all result columns.
+Which columns you receive exactly depends on the information OpenCage returns for a specific request.
+
+```{r output_all}
+oc_forward_df(placename = opera, output = "all")
+```
+
+You can also pass a data frame to `oc_forward_df()`. 
```

Comment:
```suggestion
You can also pass a data frame to `oc_forward_df()`. By default the results columns are added to the input data frame, which is useful for keeping information associated with the place names that are in separate columns. If you want a data frame with only the geocoding results, set `bind_cols = FALSE`.
```

Reply:
Just wanted a quick mention of `bind_cols` 😜

Reply:
I swear I did not leave this out on purpose! 😄